azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageVersionServlet.java

```diff
@@ -106,30 +124,98 @@ protected void handleGet(HttpServletRequest req, HttpServletResponse resp, Sessi
   }
 
   @Override
-  protected void handlePost(HttpServletRequest req, HttpServletResponse resp, Session session)
+  protected void handlePost(
+      final HttpServletRequest req, final HttpServletResponse resp, final Session session)
       throws ServletException, IOException {
+    final Map<String, String> templateVariableToValue = new HashMap<>();
+    if (BASE_IMAGE_VERSION_URI.equals(req.getRequestURI())) {
+      handleCreateImageVersion(req, resp, session);
+    } else if (SINGLE_IMAGE_VERSION_URI_TEMPLATE.match(req.getRequestURI(),
+        templateVariableToValue)) {
+      handleUpdateImageVersion(req, resp, session, templateVariableToValue);
```

Comment:
Why are we using POST for updates? PUT or PATCH are the standard HTTP methods for that.

Reply:
Will revisit in the next PR.